lib/internal/modules/esm/get_format.js

```diff
@@ -36,26 +35,25 @@ if (experimentalWasmModules)
 if (experimentalJsonModules)
   extensionFormatMap['.json'] = legacyExtensionFormatMap['.json'] = 'json';
 
-function defaultGetFormat(url, context, defaultGetFormatUnused) {
-  if (StringPrototypeStartsWith(url, 'node:')) {
-    return { format: 'builtin' };
-  }
-  const parsed = new URL(url);
-  if (parsed.protocol === 'data:') {
+const protocolHandlers = {
+  'data:'(parsed) {
     const { 1: mime } = RegExpPrototypeExec(
       /^([^/]+\/[^;,]+)(?:[^,]*?)(;base64)?,/,
       parsed.pathname,
-    ) || [ , null ];
+    ) || [, null];
     const format = ({
       '__proto__': null,
       'text/javascript': 'module',
       'application/json': experimentalJsonModules ? 'json' : null,
       'application/wasm': experimentalWasmModules ? 'wasm' : null
     })[mime] || null;
-    return { format };
-  } else if (parsed.protocol === 'file:') {
+
+    return format;
+  },
+  'file:'(parsed, url) {
```

Comment:
The `url` param here may be superfluous—it's potentially redundant to `parsed.href`, but there may be cases I haven't thought of where they are different.